packages/medusa/src/api-v2/admin/fulfillments/validators.ts

```diff
@@ -23,13 +23,15 @@ export type AdminCancelFulfillmentType = z.infer<typeof AdminCancelFulfillment>
 export const AdminCancelFulfillment = z.object({})
 
 export type AdminCreateFulfillmentType = z.infer<typeof AdminCreateFulfillment>
+// TODO: revisit the data shape this endpoint accepts
```

Comment:
question: What do we need to change?

Reply:
1. Accept either `order` or `order_id`
2. We are missing notification flag
3. Should the address be infered from the order instead of passing one?
4. Same thing for `provider_id`, isn't shipping options/provider_id already set for the order?